app/models/patient.rb

```diff
@@ -61,17 +61,17 @@ class Patient < ApplicationRecord
   scope :with_diabetes, -> { joins(:medical_history).merge(MedicalHistory.diabetes_yes) }
   scope :with_hypertension, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes) }
 
-  scope :follow_ups_by_period, ->(period, last: nil) {
-    follow_ups_with(Encounter, period, time_column: "encountered_on", last: last)
+  scope :follow_ups_by_period, ->(period, at_region: nil, current: true, last: nil) {
+    follow_ups_with(Encounter, period, at_region: at_region, current: current, time_column: "encountered_on", last: last)
   }
 
-  scope :diabetes_follow_ups_by_period, ->(period, last: nil) {
-    follow_ups_with(BloodSugar, period, last: last)
+  scope :diabetes_follow_ups_by_period, ->(period, at_region: nil, current: true, last: nil) {
+    follow_ups_with(BloodSugar, period, at_region: at_region, current: current, last: last)
       .with_diabetes
   }
 
-  scope :hypertension_follow_ups_by_period, ->(period, last: nil) {
-    follow_ups_with(BloodPressure, period, last: last)
+  scope :hypertension_follow_ups_by_period, ->(period, at_region: nil, current: true, last: nil) {
+    follow_ups_with(BloodPressure, period, at_region: at_region, current: current, last: last)
```

Comment:
Added a bit more versatility to these scopes to use them for more touchpoints now.

* `at_region`: data for BPs/BSs recorded _at_ facilities in the region (regardless of where the patient is registered/assigned)
* `current`: include/exclude the current period in the grouped results. This is just a groupdate option that we need to accept and pass down.